futures-util/src/stream/futures_unordered.rs

```diff
@@ -99,12 +103,12 @@ enum Dequeue<T> {
 }
 
 impl<T> FuturesUnordered<T>
-    where T: Future,
+    where T: Future + Unpin
```

Comment:
Here and elsewhere, `FuturesUnordered` shouldn't require that the future is `Unpin`, since it's allocating it. Instead it should track usage to ensure the future isn't moved.